src/commands/force/source/status.ts

```diff
@@ -5,55 +5,70 @@
 import { trackingSetup } from '../../../trackingFunctions';
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/plugin-source', 'status');
 
-export default class Status extends SfdxCommand {
+export type StatusCommandResult = StatusResult[];
+
+export default class Status extends SfCommand<StatusCommandResult> {
+  public static readonly deprecateAliases = true;
   public static aliases = ['force:source:beta:status'];
-  public static description = messages.getMessage('description');
-  public static readonly examples = messages.getMessage('examples').split(os.EOL);
-  protected static flagsConfig: FlagsConfig = {
-    local: flags.boolean({
+  public static readonly summary = messages.getMessage('description');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly flags = {
+    'api-version': orgApiVersionFlagWithDeprecations,
+    loglevel,
+    'target-org': requiredOrgFlagWithDeprecations,
+    local: Flags.boolean({
       char: 'l',
       description: messages.getMessage('flags.local'),
-      longDescription: messages.getMessage('flags.localLong'),
+      summary: messages.getMessage('flags.localLong'),
       exclusive: ['remote'],
     }),
-    remote: flags.boolean({
+    remote: Flags.boolean({
       char: 'r',
       description: messages.getMessage('flags.remote'),
-      longDescription: messages.getMessage('flags.remoteLong'),
+      summary: messages.getMessage('flags.remoteLong'),
       exclusive: ['local'],
     }),
-    concise: flags.builtin({
-      description: messages.getMessage('flags.concise'),
+    concise: Flags.boolean({
+      summary: messages.getMessage('flags.concise'),
     }),
   };
-  protected static requiresUsername = true;
-  protected static requiresProject = true;
+  public static readonly requiresProject = true;
   protected results = new Array<StatusResult>();
   protected localAdds: ChangeResult[] = [];
+  private flags: Interfaces.InferredFlags<typeof Status.flags>;
 
-  public async run(): Promise<StatusResult[]> {
+  public async run(): Promise<StatusCommandResult> {
+    this.flags = (await this.parse(Status)).flags;
     const tracking = await trackingSetup({
       commandName: 'force:source:status',
       ignoreConflicts: true,
-      org: this.org,
+      org: this.flags['target-org'],
       project: this.project,
-      ux: this.ux,
+      ux: new Ux({ jsonEnabled: this.jsonEnabled() }),
     });
 
-    const wantsLocal = (this.flags.local as boolean) || (!this.flags.remote && !this.flags.local);
-    const wantsRemote = (this.flags.remote as boolean) || (!this.flags.remote && !this.flags.local);
+    const wantsLocal = this.flags.local || (!this.flags.remote && !this.flags.local);
+    const wantsRemote = this.flags.remote || (!this.flags.remote && !this.flags.local);
 
-    this.logger.debug(
+    this.debug(
```

Comment:
I'm really intrigued by this pattern and where these end up.  Is oclif/command/debug sending them where you'd expect?